dvc/command/gc.py

```diff
@@ -9,19 +9,28 @@
 
 logger = logging.getLogger(__name__)
 
+supported_removal_filters = ("commits", "branches", "tags")
+
 
 class CmdGC(CmdBase):
     def run(self):
-        msg = "This will remove all cache except items used in "
 
+        msg = "This will remove all cache except items used in "
         msg += "the working tree"
-        if self.args.all_commits:
+
+        self.args.remove = self.args.remove or []
+
+        if "commits" not in self.args.remove:
             msg += " and all git commits"
-        elif self.args.all_branches and self.args.all_tags:
+
+        if (
+            "branches" not in self.args.remove
+            and "tags" not in self.args.remove
```

Comment:
Maybe make `self.args.remove` a set:
```python
if not {"branches", "tags"} & self.args.remove:
    # ...
```
```

Reply:
Please don't nitpick at the moment. This PR does not work. It's a WIP for a reason.